server/lib/nl/detection/llm_detector.py

```diff
@@ -200,37 +212,33 @@ def detect(query: str, context_history: Dict, index_type: str,
         continue
       cls = _handle_compare(llm_vals[0])
     elif t == filter_type:
-      # Earlier since we had set filter_type, should be safe to do llm_resp[f][0].
-      cls = _handle_quantity(llm_resp[f][0], t)
+      # Earlier caller had set filter_type by checking llm_resp,
+      # should be safe to do llm_resp[t][0].
+      cls = _handle_quantity(llm_resp[t][0], t)
```

Comment:
"cls" is a python reserved word, maybe use some other name? (i was a little confused when first see this)

Reply:
Done.